processing/src/main/java/io/druid/segment/StringDimensionMergerV9.java

```diff
@@ -120,17 +122,19 @@ public void writeMergedValueMetadata(List<IndexableAdapter> adapters) throws IOE
     long dimStartTime = System.currentTimeMillis();
 
     this.adapters = adapters;
+    this.dimValuesList = toDimValueList(adapters);
```

Comment:
this is some cleanup to StringDimensionMergerV9.java, in this part, only dimValuesList is needed, but more info is required(the adapters) than needed.